### ISSUE TITLE
Improve support for self installation of Pods

### ISSUE THREAD

**Issue by orta** [CocoaPods/CocoaPods#3589]
Given that a lot of our raised issues revolve around the installation and injection of Pods into the workspace and ensuring Xcode does everything for all the edge cases. It might make sense to do offer a Carthage style option. It could work by you declaring in the podfile that you want to do a self-installation. We can upfront say that this only works frameworks. More or less what's going on with Rome.

This generates a `Pods` directory but instead of all the workspaces it includes the separated compiled frameworks.

Podfile:
```
use_frameworks!
self_installation!

pod "AFNetworking"
pod "ORStackView"
```

You run `pod install` and this generates

```
Energy $ ls Pods 
AFNetworking.framework                        ORStackView.framework
```

Ideally: running this in a project with an existing integrated workspace offers to de-integrate via `pod deintegrate`.

**Comment by neonichu** [comment]
If it only works with frameworks anyway, I'd say `self_installation!` could imply `use_frameworks!`. 

Two other questions regarding platforms:

- What to do when one project needs both Mac and iOS frameworks.

- We need to offer a way to strip out Intel slices for iOS App Store submissions. Carthage was facing the same problem and now has a specific subcommand for that which they recommend running via a build-phase. Since we don't want projects to depend on CP, we'd probably need to generate a shell-script in the Pods directory

**Comment by orta** [comment]
> I'd say self_installation! could imply use_frameworks!.

That seems reasonable.

> We need to offer a way to strip out Intel slices for iOS App Store submissions.

Could we generate both on `pod install`, then have the stripped ones in a folder with a readme?

**Comment by neonichu** [comment]
> Could we generate both on pod install, then have the stripped ones in a folder with a readme?

Generally, yes, but there's no good way to integrate them without creating a second target for simulator builds. I'd expect that people want to switch between simulator and device builds of their app, just like when integrating with CP.

**Comment by orta** [comment]
> I'd expect that people want to switch between simulator and device builds of their app, just like when integrating with CP.

Xcode plugin? ;)

Can we add _recommend_ adding a step specifically to the archive build phase?

**Comment by neonichu** [comment]
Carthage seems to use the `VALID_ARCHS` env variable, I guess that will give us Intel on sim builds and ARM for device ones: <https://github.com/Carthage/Carthage/blob/master/Source/carthage/CopyFrameworks.swift#L75>

**Comment by orta** [comment]
So the tradeoff for using this kind of architecture is that you have to have CP installed to de-sym the frameworks in the build phases when it changes?

**Comment by neonichu** [comment]
I'd try to avoid needing an actual CP installation by putting the symbol stripping into a script, similar like "Copy Resources", etc. right now. We'd then advise users to integrate that script in a build phase.

**Comment by orta** [comment]
At what point would they be stripped? Would switching devices in Xcode "just work" without a `pod install`

**Comment by neonichu** [comment]
Yah, they'd be stripped during the build, not touching the originals.

**Comment by orta** [comment]
Sounds like a trade-off we have no choice in.

So what's the thing about Mac/iOS frameworks?

**Comment by neonichu** [comment]
So, Rome builds whatever platform was chosen in the Podfile, but that can only be one and users have to chose when using `--no-integrate`. So I'd say we build both platforms by default and in case one was explicitly chosen just one.

**Comment by orta** [comment]
:+1:

**Comment by darrenclark** [comment]
I'd really love this option!

Also, it'd be pretty slick if Cocoapods could generate a Xcode project with a framework target for Development Pods that I could drag into my workspace.  If that makes sense? So we'd end up with something like (for example):

```
Pods/
     AFNetworking.framework
     ReactiveCocoa.framework
     MyInternalPod.xcodeproj
```

That way development pods would still be easily editable.

**Comment by brentleyjones** [comment]
I would love this option. It's the main reason I'm personally a fan of Carthage right now.

**Comment by neonichu** [comment]
@brentleyjones would be great to see a PR for it!

**Comment by brentleyjones** [comment]
@neonichu I would if I could, but I only know Swift and Objective-C. Plus most of CocoaPods is black magic to me, which is why I would love this option. It removes a bit of the *magic*. I'm just here to give my :+1: :smile:.

**Comment by neonichu** [comment]
@brentleyjones we're always happy to help new contributors along, most of us didn't know Ruby or CocoaPods very well before we started. You might check out <http://blog.cocoapods.org/starting-open-source/> and reconsider :)

**Comment by alexargo** [comment]
Seems like @neonichu has been busy with his plugin in the past few months, but it hasn't been updated  that recently (maybe it has been super stable). Are there plans to integrate it into cocoapods proper or keep it seperate, or is it the project on hiatus? Definitely would love to use this, now that I'm working on dropping support for iOS 7 and will be able to use frameworks.

**Comment by orta** [comment]
It's feeling like this feature might make it into the next major release after 0.39, hard to say without making promises.

**Comment by andreyvit** [comment]
Are we only discussing a Carthage-style binary frameworks option here?

I believe that a better (or at least a worthwhile) alternative is to just make xcodeproj files for those frameworks and allow Xcode to build them properly. This avoids the problem of architecture stripping, and generally it's always better to commit sources than binaries (assuming that you're committing the Pods folder).

**Comment by neonichu** [comment]
I'm closing this in favour of neonichu/Rome#32 :)
